### ISSUE TITLE
Enhancement: close half-open braces, use the snippet system to smartly place the point

### ISSUE THREAD

**Issue by Lucretiel** [zxqfl/tabnine-sublime#3]
Suppose I'm creating a new rust struct, with some derive attributes at the top. I type:

```
#
```

And TabNine suggests `[derive(Clone,  `.

This is a good suggestion, but I'm very much used to my editor automatically closing my braces. What I'm suggesting, therefore, is that the following flow occur (█ indicates the suggestion of the point; I'm not sure how to indicate this in the UI. Perhaps a highlighted character?)

- TabNine suggests `[derive(Clone, █)]`
- I accept the suggestion. My text buffer is now:

```
#[derive(Clone, █)]
```

- I have the ability to add more attributes, aided by TabNine.
- The next time I hit \<tab\>, outside of a completion context, it advances the point outside the next brace. It does this continuously until I'm fully outside the braces:

```
// <tab>
#[derive(Clone, Default)█]

// <tab>
#[derive(Clone, Default)]█
```

**Comment by zxqfl** [comment]
Closing of half-open braces has been added. Snippet support is still pending. (Supporting the workflow you describe requires a rethinking of the current interface, since pressing tab after a completion currently selects the next completion.)

**Comment by Lucretiel** [comment]
Yeah, I was thinking about that, too. You'd have to do some extra typing to
get out of the tabnine context. Still, glad to hear that brace closing
works!
